watch_service.go

```diff
@@ -287,13 +279,11 @@ func (w *Watcher) scheduleWatch(numAttemptsSoFar int) {
 		delay = nextDelay(numAttemptsSoFar)
 	}
 
-	go func() {
-		select {
-		case <-w.watchCTX.Done():
-		case <-time.NewTimer(delay).C:
-			w.doWatch(numAttemptsSoFar)
-		}
-	}()
+	select {
```

Comment:
Using
```
for {
  select{
  }
}
```
Instead of recursion calling between `doWatch` and `scheduleWatch` is much more better. 
`numAttemptsSoFar` could be a field in `Watcher` and shared between 2 function